src/main/java/slimeknights/tconstruct/library/tools/ToolNBT.java

```diff
@@ -12,13 +12,15 @@
   public int harvestLevel;
   public float attack;
   public float speed; // mining speed
+  public float attackSpeed;
   public int modifiers; // free modifiers
 
   public ToolNBT() {
     durability = 0;
     harvestLevel = 0;
     attack = 0;
     speed = 0;
+    attackSpeed = 0;
```

Comment:
Initialize with 1 here, remove assignment of 1 in all the tool data building. As long as there is no other value set by something else it should always be 1. Also rename it to attackSpeedMultiplier, to not confuse it with the actual attackspeed stat of the tool (which is the product of the value + the tools AS)